VmAgent.Core/Interfaces/ProcessRunner.cs

```diff
@@ -66,16 +72,46 @@ public override Task<SessionHostInfo> CreateAndStart(int instanceNumber, GameRes
                 _logger.LogException($"Failed to start process based host with instance number {instanceNumber}", exception);
                 sessionHostManager.RemoveSessionHost(sessionHostUniqueId);
                 sessionHost = null;
+                string exceptionMessage = exception.ToString();
+                CollectLogs(logFolderPathOnVm, exceptionMessage);
             }
 
             return Task.FromResult(sessionHost);
         }
 
-        public override Task CollectLogs(string processId, string logsFolder, ISessionHostManager sessionHostManager)
+        public override Task CollectLogs(string id, string logsFolder, ISessionHostManager sessionHostManager)
+        {
+            return Task.CompletedTask;
+        }
+
+        public override Task CollectLogs(string logsFolder, string exceptionMessage)
         {
             // The game server is free to read the env variable for log folder and write all output to a file in that folder.
             // If required, we can add action handlers (see SystemOperations.RunProcess for example).
             // However, keeping the file handle around can be tricky.
+            if (_systemOperations.IsOSPlatform(OSPlatform.Windows))
+            {
+                try
+                {
+                    _logger.LogVerbose("Collecting logs for failed start game process.");
+                    string destinationFileName = Path.Combine(logsFolder, ConsoleLogCaptureFileName);
+
+                    using (StreamWriter sw = File.AppendText(destinationFileName))
+                    {
+                        sw.WriteLine("{0} {1}", DateTime.Now.ToLongTimeString(), DateTime.Now.ToLongDateString());
```

Comment:
>ToLongTimeString [](http://example.com/codeflow?start=61&length=16)

You can consolidate these to calls into a single one using:
DateTime.Now.ToString(" < enter format here > ")

You can see the available formats and examples here: 
[https://docs.microsoft.com/en-us/dotnet/standard/base-types/custom-date-and-time-format-strings](https://docs.microsoft.com/en-us/dotnet/standard/base-types/custom-date-and-time-format-strings)